cyvlfeat/fisher/cyfisher.pyx

```diff
@@ -59,24 +82,57 @@ cpdef cy_fisher(float[:, :] X,
             n_data, n_clusters, n_dimensions, 2 * n_clusters * n_dimensions,
             square_root, normalized, fast))
 
-    cdef float[:] enc = np.zeros(2 * n_clusters * n_dimensions,
-                                 dtype=np.float32)
+    if X.dtype == "float32":
+
+        encf = np.zeros(
+            2 * n_clusters * n_dimensions, dtype=X.dtype
+        )
+        meansf = MEANS
+        covarsf = COVARIANCES
+        priorsf = PRIORS
+        Xf = X
+
+        num_terms = vl_fisher_encode(&encf[0],
```

Comment:
Having looked at the underyling C code I agree that this ordering is consistent given the swapping of the `n_dimensions` and `n_clusters`.